### ISSUE TITLE
Catch eventual errors from case search page

### ISSUE THREAD

**Issue by northwestwitch** [Clinical-Genomics/scout#3316]
```
2022-04-26 14:28:32,362 - [scout.server.app](http://scout.server.app/) - ERROR: Exception on /api/v1/hpo-terms [GET] [in /venv/lib/python3.8/site-packages/flask/app.py:1455]
Traceback (most recent call last):
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/venv/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "/venv/lib/python3.8/site-packages/flask/app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/home/worker/app/scout/server/blueprints/cases/views.py", line 553, in hpoterms
    terms = sorted(store.hpo_terms(query=query), key=itemgetter("hpo_number"))
  File "/venv/lib/python3.8/site-packages/pymongo/cursor.py", line 1238, in next
    if len(self.__data) or self._refresh():
  File "/venv/lib/python3.8/site-packages/pymongo/cursor.py", line 1155, in _refresh
    self.__send_message(q)
  File "/venv/lib/python3.8/site-packages/pymongo/cursor.py", line 1044, in __send_message
    response = client._run_operation(
  File "/venv/lib/python3.8/site-packages/pymongo/mongo_client.py", line 1424, in _run_operation
    return self._retryable_read(
  File "/venv/lib/python3.8/site-packages/pymongo/mongo_client.py", line 1525, in _retryable_read
    return func(session, server, sock_info, secondary_ok)
  File "/venv/lib/python3.8/site-packages/pymongo/mongo_client.py", line 1420, in _cmd
    return server.run_operation(
  File "/venv/lib/python3.8/site-packages/pymongo/server.py", line 130, in run_operation
    _check_command_response(first, sock_info.max_wire_version)
  File "/venv/lib/python3.8/site-packages/pymongo/helpers.py", line 167, in _check_command_response
    raise OperationFailure(errmsg, code, response, max_wire_version)
pymongo.errors.OperationFailure: Regular expression is invalid: unmatched parentheses, full error: {'operationTime': Timestamp(1650976112, 8), 'ok': 0.0, 'errmsg': 'Regular expression is invalid: unmatched parentheses', 'code': 51091, 'codeName': 'Location51091', '$clusterTime': {'clusterTime': Timestamp(1650976112, 8), 'signature': {'hash': b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00', 'keyId': 0}}}
```

**Comment by mikaell** [comment]
After discussion with @northwestwitch, we suspect error fixed in another PR.

**Comment by northwestwitch** [comment]
This was fixed by #3356
